newsroom/search.py

```diff
@@ -528,11 +528,15 @@ def prefill_search_highlights(self, search, req):
                     "lenient": True,
                 },
             )
-            elastic_highlight_query["fields"] = {
-                "body_html": field_settings,
-                "headline": field_settings,
-                "slugline": field_settings,
-            }
+            selected_fields = json.loads(advanced_search).get("fields") if advanced_search else []
```

Comment:
imo it would be good to do that `json.loads` thing once in the beginning and not each time `advanced_search` is used for something